Mlem/API/APIClient.swift

```diff
@@ -91,4 +51,31 @@ class APIClient {
             throw APIClientError.networking(error)
         }
     }
+    
+    private func urlRequest(from defintion: any APIRequest) throws -> URLRequest {
+        var urlRequest = URLRequest(url: defintion.endpoint)
+        defintion.headers.forEach { header in
+            urlRequest.setValue(header.value, forHTTPHeaderField: header.key)
+        }
+        
+        if let _ = defintion as? any APIGetRequest {
+            urlRequest.httpMethod = "GET"
+        } else if let postDefinition = defintion as? any APIPostRequest {
+            urlRequest.httpMethod = "POST"
+            urlRequest.httpBody = try createBodyData(for: postDefinition)
+        } else if let putDefinition = defintion as? any APIPutRequest {
```

Comment:
`APIPutRequest` is defined as a `APIPostRequest` subclass, wont the above selection (`as? PostRequest`) catch all put requests?

Reply:
Somehow this comment duplicated? huh